### PR TITLE
Revert "ci: Dont remove packages in ci jobs (#17146)"

### PR DESCRIPTION
**Pull request by phlax**
This reverts commit e330009658366a518c54ee747caca0a5dae6abe9.

Signed-off-by: Ryan Northey <ryan@synca.io>

Commit Message:
Additional Description:

I added this previously as it was slowing down ci and it didnt seem necessary - but for #19275 i think we need to revert

i have some additional space saving that we could do, but tbh this is probably enough

Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by phlax** [comment]
cc @PiotrSikora @lizan

File: .azure-pipelines/bazel.yml [hunk 1]
```diff
@@ -25,6 +25,10 @@ steps:
     path: $(Build.StagingDirectory)/repository_cache
   continueOnError: true
 
+- bash: .azure-pipelines/cleanup.sh
```
Comment (phlax) [inline .azure-pipelines/bazel.yml:28]: i dont think this is necesary just the docker one

File: .azure-pipelines/cleanup.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+set -e
+
+# Temporary script to remove tools from Azure pipelines agent to create more disk space room.
+sudo apt-get update -y || true
+sudo apt-get purge -y --no-upgrade 'ghc-*' 'zulu-*-azure-jdk' 'libllvm*' 'mysql-*' 'dotnet-*' 'libgl1' \
+  'adoptopenjdk-*' 'azure-cli' 'google-chrome-stable' 'firefox' 'hhvm'
```
Comment (lizan) [inline .azure-pipelines/cleanup.sh:8]: From CI log:
```481078	google-cloud-sdk
317653	temurin-17-jdk
309543	temurin-11-jdk
214576	mongodb-mongosh
195844	temurin-8-jdk
188266	powershell
```

Let's add `temurin-*-jdk` here as well?

Comment (PiotrSikora) [reply .azure-pipelines/cleanup.sh:8]: +1

Maybe purge `mongodb-mongosh` as well, I doubt it's being used on the CI.

**Review by lizan — APPROVED** [review]

**Pull request merged**